cmd/client-s3.go

```diff
@@ -1012,12 +1012,20 @@ func (c *s3Client) MakeBucket(region string, ignoreExisting, withLock bool) *pro
 		return probe.NewError(BucketNameEmpty{})
 	}
 	if object != "" {
-		if strings.HasSuffix(object, "/") {
+		if !strings.HasSuffix(object, string(c.targetURL.Separator)) {
```

Comment:
I had a hard time understanding this, so I tried to rewrite it so return paths are more clear:
```Go
	if object != "" {
		if !strings.HasSuffix(object, string(c.targetURL.Separator)) {
			object = path.Dir(object)
		}
		if !strings.HasSuffix(object, string(c.targetURL.Separator)) {
			return probe.NewError(BucketNameTopLevel{})
		}
		var retried bool
		for {
			_, e := c.api.PutObject(bucket, object, bytes.NewReader([]byte("")), 0, minio.PutObjectOptions{})
			if e != nil {
				if retried {
					return probe.NewError(e)
				}
				switch minio.ToErrorResponse(e).Code {
				case "NoSuchBucket":
					if withLock {
						e = c.api.MakeBucketWithObjectLock(bucket, region)
					} else {
						e = c.api.MakeBucket(bucket, region)
					}
					if e != nil {
						return probe.NewError(e)
					}
					retried = true
					continue
				}
				return probe.NewError(e)
			}
			return nil
		}
	}
```
Also added a single retry only, so we don't risk an infinite loop if something is broken.